forte/model.py

```diff
@@ -0,0 +1,109 @@
+from abc import ABC, abstractmethod
+from forte.molecule import Molecule
+from forte.basis import Basis
+from forte.forte import StateInfo
+from forte.forte import Symmetry
+
+
+class Model(ABC):
+    """
+    An abstract base class used to implement a model.
+
+    Attributes
+    ----------
+    """
+    @abstractmethod
+    def point_group(self) -> str:
+        """The model point group"""
+
+
+class MolecularModel(Model):
+    """
+    A class used to handle molecules.
+
+    This class is used to compute the integrals once a model has been defined.
+
+    Attributes
+    ----------
+    """
+    def __init__(self, molecule: Molecule, basis: Basis, scf_aux_basis: Basis = None, corr_aux_basis: Basis = None):
+        self._molecule = molecule
+        self._basis = basis
+        self._scf_aux_basis = scf_aux_basis
+        self._corr_aux_basis = corr_aux_basis
+        self.symmetry = Symmetry(molecule.molecule.point_group().symbol().capitalize())
+
+    def __repr__(self):
+        """
+        return a string representation of this object
+        """
+
+    def __str__(self):
+        """
+        return a string representation of this object
+        """
+
+    @property
+    def molecule(self):
+        return self._molecule.molecule
+
+    @property
+    def basis(self):
+        return self._basis.basis
+
+    @property
+    def scf_aux_basis(self):
+        if self._scf_aux_basis is None:
+            return None
+        return self._scf_aux_basis.basis
+
+    @property
+    def corr_aux_basis(self):
+        if self._corr_aux_basis is None:
+            return None
+        return self._corr_aux_basis.basis
+
+    @property
+    def point_group(self) -> str:
+        return self.symmetry.point_group_label()
+
+    def state(self, charge: int, multiplicity: int, ms: float = None, sym: str = None):
+        """This function is used to create a StateInfo object. It checks for potential errors."""
```

Comment:
The meaning of these arguments should be documented.

Reply:
Documented.